packages/cli-repl/src/cli-repl.ts

```diff
@@ -68,6 +75,48 @@ class CliRepl {
       this.connect(driverUri, driverOptions);
     }
   }
+
+  /**
+  * Creates a directory to store all mongosh logs, history and config
+  */
+  createMongoshDir(): void {
+    try {
+      mkdirp.sync(this.mongoshDir);
+    } catch(e) {
+      this.bus.emit('error', e)
+      throw e;
+    }
+  }
+
+  /**
+  * Checks if config file exists.
+  *
+  * If exists: sets userId and enabledTelemetry to this.
+  * If does not exist: writes a new file with a newly generated ObjectID for
+  * userid and enableTelemetry set to false.
+  */
+  generateOrReadTelemetryConfig(): void {
```

Comment:
How do we sync `fs.open` here?

Reply:
like if `fs.open` would take long enough (or it would hang)

Reply:
Both `fs.open` and `fs.openSync` could be blocking or take long enough, the only difference (iirc) is that `fs.open` is backed by a thread pool. If you are worried about it taking too long, I can look into how to add a timeout. 

I did look at having it done synchronously, and here is what that looked like:

```js
    let fd;

    try {
      fd = fs.openSync(configPath, 'wx');

      this.userId = new ObjectId(Date.now());
      this.enableTelemetry = false;
      const config = { userId: this.userId, enableTelemetry: this.enableTelemetry };
      fs.writeFileSync(fd, JSON.stringify(config));
    } catch (err) {
      if (err.code === 'EEXIST') {
        const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
        this.userId = config.userId;
        this.enableTelemetry = config.enableTelemetry;
        return;
      }
      this.bus.emit('error', e)
      throw err;

    } finally {
      if (fd !== undefined) fs.closeSync(fd);
    }
```

Reply:
I meant that with the async version we could have race conditions. Like `this.userId not being set when we try to read it.

Reply:
a race condition is a horse of a different colour! i can make it sync.